CommandDotNet.Tests/Parsing/SingleValueParserTests.cs

```diff
@@ -32,74 +32,29 @@ public class SingleValueParserTests
                 {"Decimal", typeof(decimal), "4.5", 4.5m},
                 {"Person", typeof(Person), "lala", new Person("lala")},
             };
-    
-        public class PropertyModel
-        {
-            public int Int { get; set; }
-            public string String { get; set; }
-            public double Double { get; set; }
-            public long Long { get; set; }
-            public bool Bool { get; set; }
-            public Time Time { get; set; }
-            public char Char { get; set; }
-            public decimal Decimal { get; set; }
-            public Person Person { get; set; }
-        }
-        
+
         [Theory]
         [MemberData(nameof(Values))]
-        public void CanParseValues(string propertyName, Type valueType, string value, object typedValue)
+        public void CanParseValues(string propertyName, Type valueType, string stringValue, object typedValue)
         {
-            var typeDescriptor = _appSettings.ArgumentTypeDescriptors.GetDescriptor(valueType);
-            var valueParser = new SingleValueParser(typeDescriptor);
+            var commandOptionInfo = _commandOptionInfos[propertyName];
+            commandOptionInfo.SetValueForTest(stringValue);
+
+            var valueParser = new ParserFactory(_appSettings).CreateInstance(commandOptionInfo);
+            valueParser.Should().BeOfType<SingleValueParser>();
 
-            var commandOptionInfo = _commandOptionInfos[propertyName].SetValueForTest(value);
-            commandOptionInfo.SetValueForTest(value);
-            
             object parsedValue = valueParser.Parse(commandOptionInfo);
 
-            parsedValue.Should().BeOfType(valueType);
             parsedValue.Should().Be(typedValue);
-        }
-
-        private static AppSettings InitAppSettings()
-        {
-            var appSettings = new AppSettings();
-            appSettings.ArgumentTypeDescriptors.Add(new PersonTypeDescriptor());
-            return appSettings;
-        }
-
-        public class Person
-        {
-            public string Name { get; }
 
-            public Person(string name)
+            var underlyingType = Nullable.GetUnderlyingType(valueType);
```

Comment:
do we need to validate the type here?  i think the previous equality assertion should be adequate and then we can remove these lines.

Reply:
I guess it will make a difference when comparing int 3 with double 3.0. We want the test to fail in that case.With just value comparison, test will pass I think..

Reply:
yeah, good catch.  thanks